org.sailfishos.usermanager.xml

```diff
@@ -28,6 +28,15 @@
         <arg type ="u" name="uid"/>
         <arg type ="s" name="new_full_name"/>
     </signal>
+    <method name="switchUser">
+        <arg direction="in" type="u" name="uid"/>
+    </method>
+    <signal name="userSwitched">
+        <arg type ="u" name="uid"/>
+    </signal>
+    <signal name="userSwitchFailed">
+        <arg type ="u" name="uid"/>
+    </signal>
   </interface>
```

Comment:
I would suggest a `currentUser` property (or `getCurrentUser` call if you prefer it that way), and rename `userSwitched` into `currentUserChanged`.

`userSwitchFailed` sounds ok.

Reply:
There also should be a property that would tell clients that user is being switched, I think. That can be done in a number of ways, for example by introducing `nextUser` or `userSwitchingTo` property (or `getNextUser` call or something like that) together with `nextUserChanged` signal. Then if `currentUser != nextUser`, then user switch is in progress.

Reply:
Added currentUser method but keeping userSwithed as it better names it being response to userSwitch.

Reply:
I won't argue about naming, that's a very subjective topic. But don't think about it as a response. Response is sent directly to the caller and it's a separate type of D-Bus message. This is a signal, its recipients don't necessarily know and don't have to to know what's caused it.